---
title: "Results from `R/qtl2` Analayis (Plots) - max IOP [all / 69k grid]"
author: "Belinda Cornes"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#pheno
name.phenocsv = "DO-Lmx1b"
setwd("/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/workflowr/John-Tolman_Workflow/")

pheno <- read.csv("/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/Data/meta.data/DO-Lmx1b_pheno.covar.csv")

library(qtl2)
library(ggplot2)

#map files
#gmap.file = qtl2::read_csv("data/combined/genetic_map_sorted.csv")
#pmap.file = qtl2::read_csv("data/combined/physical_map_sorted.csv")

#This calls reorder_map_table() and the code for that is:
#reorder_map_table <- function (map_tab, chr_col = 1, pos_col = 2, chr_names = NULL)
#{
#  chr <- map_tab[, chr_col]
#  if (is.null(chr_names))
#    chr_names <- unique(chr)
#  chr <- factor(chr, levels = chr_names)
#  pos <- suppressWarnings(as.numeric(map_tab[, pos_col]))
#  map_tab[order(chr, pos, seq_len(nrow(map_tab))), , drop = FALSE]
#}

#split_map <- function (map, chr_names = NULL)
#{
#  map <- reorder_map_table(map, chr_names = chr_names)
#  pos <- as.numeric(map[, 2])
#  chr <- map[, 1]
#  uchr <- unique(chr)
#  names(pos) <- rownames(map)
#  lapply(split(pos, factor(chr, uchr)), sort)
#}

#gmap <- split_map(gmap.file)
#pmap <- split_map(pmap.file)

#apr <- readRDS("/Users/corneb/Box/Simon.John-Nick.Tolman_Data/2020-12-28/combined_v2/Jax.Columbia_John__GigaMUGA_genoprobs_8state_69kchr_sorted.rds")
map <- readRDS("/Users/corneb/Documents/MyJax/CS/Projects/QTL2_support.files/marker_grid_0.02cM_plus_map.rds")

```

This script will plot the qtl mapping results.

```{r phenotype, message=FALSE}

i = names(pheno)[13]
i


```

### Plot Scan Results: m1 (Sex + Generation)
```{r plot scan result for m1, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Generation (m1)")
load(paste0("output/qtl.mapping_sorted_all.grid/m1_sex.generation_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m1.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m1_sex.generation_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m1_sex.generation_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m1.qtl.out[[i]]) # overall maximum LOD score
plot(m1.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & generation) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m1_sex.generation_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m1.gwas[[i]]$lod, 
     m1.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & generation) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m1_sex.generation_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m1_sex.generation_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m1.qtl.out)){
  chr <- as.character(m1.peak[[i]][m1.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m1.coef[[i]], 
              map[chr], 
              scan1_output=m1.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation) (scan1coef): ",i)
              )
  plot_coefCC(m1.blup[[i]], 
              map[chr], 
              scan1_output=m1.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation) [blup values] (scan1blup): ",i)
              )
  plot(m1.snps[[i]]$lod, 
       m1.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m1.genes[[i]],
       main = paste0("SNP Association Mapping (sex & generation) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m2 (Sex)
```{r plot scan result for m2,fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex (m2)")
load(paste0("output/qtl.mapping_sorted_all.grid/m2_sex_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m2.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m2_sex_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m2_sex_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m2.qtl.out[[i]]) # overall maximum LOD score
plot(m2.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m2_sex_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m2.gwas[[i]]$lod, 
     m2.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m2_sex_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m2_sex_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m2.qtl.out)){
  chr <- as.character(m2.peak[[i]][m2.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m2.coef[[i]], 
              map[chr], 
              scan1_output=m2.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex) (scan1coef): ",i)
              )
  plot_coefCC(m2.blup[[i]], 
              map[chr], 
              scan1_output=m2.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex) [blup values] (scan1blup): ",i)
              )
  plot(m2.snps[[i]]$lod, 
       m2.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m2.genes[[i]],
       main = paste0("SNP Association Mapping (sex) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m3 (Sex + Generation + Diet)
```{r plot scan result for m3,fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Generation & Diet (m3)")
load(paste0("output/qtl.mapping_sorted_all.grid/m3_sex.generation.diet_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m3.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m3_sex.generation.diet_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m3_sex.generation.diet_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m3.qtl.out[[i]]) # overall maximum LOD score
plot(m3.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & generation & diet) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m3_sex.generation.diet_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m3.gwas[[i]]$lod, 
     m3.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & generation & diet) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m3_sex.generation.diet_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m3_sex.generation.diet_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m3.qtl.out)){
  chr <- as.character(m3.peak[[i]][m3.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m3.coef[[i]], 
              map[chr], 
              scan1_output=m3.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & diet) (scan1coef): ",i)
              )
  plot_coefCC(m3.blup[[i]], 
              map[chr], 
              scan1_output=m3.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & diet) [blup values] (scan1blup): ",i)
              )
  plot(m3.snps[[i]]$lod, 
       m3.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m3.genes[[i]],
       main = paste0("SNP Association Mapping (sex & generation & diet) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m4 (Sex + Diet)
```{r plot scan result for m4,fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Diet (m4)")
load(paste0("output/qtl.mapping_sorted_all.grid/m4_sex.diet_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m4.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m4_sex.diet_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m4_sex.diet_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m4.qtl.out[[i]]) # overall maximum LOD score
plot(m4.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & diet) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m4_sex.diet_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m4.gwas[[i]]$lod, 
     m4.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & diet) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m4_sex.diet_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m4_sex.diet_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m4.qtl.out)){
  chr <- as.character(m4.peak[[i]][m4.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m4.coef[[i]], 
              map[chr], 
              scan1_output=m4.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & diet) (scan1coef): ",i)
              )
  plot_coefCC(m4.blup[[i]], 
              map[chr], 
              scan1_output=m4.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & diet) [blup values] (scan1blup): ",i)
              )
  plot(m4.snps[[i]]$lod, 
       m4.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m4.genes[[i]],
       main = paste0("SNP Association Mapping (sex & diet) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m5 (Sex + Generation + Strain)
```{r plot scan result for m5,fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Generation & Strain (m5)")
load(paste0("output/qtl.mapping_sorted_all.grid/m5_sex.generation.strain_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m5.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m5_sex.generation.strain_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m5_sex.generation.strain_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m5.qtl.out[[i]]) # overall maximum LOD score
plot(m5.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & generation & strain) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m5_sex.generation.strain_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m5.gwas[[i]]$lod, 
     m5.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & generation & strain) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m5_sex.generation.strain_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m5_sex.generation.strain_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m5.qtl.out)){
  chr <- as.character(m5.peak[[i]][m5.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m5.coef[[i]], 
              map[chr], 
              scan1_output=m5.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & strain) (scan1coef): ",i)
              )
  plot_coefCC(m5.blup[[i]], 
              map[chr], 
              scan1_output=m5.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & strain) [blup values] (scan1blup): ",i)
              )
  plot(m5.snps[[i]]$lod, 
       m5.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m5.genes[[i]],
       main = paste0("SNP Association Mapping (sex & generation & strain) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m6 (Sex + Strain)
```{r plot scan result for m6,fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Strain (m6)")
load(paste0("output/qtl.mapping_sorted_all.grid/m6_sex.strain_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m6.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m6_sex.strain_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m6_sex.strain_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m6.qtl.out[[i]]) # overall maximum LOD score
plot(m6.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & strain) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m6_sex.strain_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m6.gwas[[i]]$lod, 
     m6.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & strain) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m6_sex.strain_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m6_sex.strain_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m6.qtl.out)){
  chr <- as.character(m6.peak[[i]][m6.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m6.coef[[i]], 
              map[chr], 
              scan1_output=m6.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & strain) (scan1coef): ",i)
              )
  plot_coefCC(m6.blup[[i]], 
              map[chr], 
              scan1_output=m6.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & strain) [blup values] (scan1blup): ",i)
              )
  plot(m6.snps[[i]]$lod, 
       m6.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m6.genes[[i]],
       main = paste0("SNP Association Mapping (sex & strain) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m7 (Sex + Generation + Diet + Strain)
```{r plot scan result for m7,fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Generation & Diet & Strain (m7)")
load(paste0("output/qtl.mapping_sorted_all.grid/m7_sex.generation.diet.strain_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m7.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m7_sex.generation.diet.strain_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m7_sex.generation.diet.strain_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m7.qtl.out[[i]]) # overall maximum LOD score
plot(m7.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & generation & diet & strain) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m7_sex.generation.diet.strain_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m7.gwas[[i]]$lod, 
     m7.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & generation & diet & strain) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m7_sex.generation.diet.strain_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m7_sex.generation.diet.strain_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m7.qtl.out)){
  chr <- as.character(m7.peak[[i]][m7.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m7.coef[[i]], 
              map[chr], 
              scan1_output=m7.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & diet & strain) (scan1coef): ",i)
              )
  plot_coefCC(m7.blup[[i]], 
              map[chr], 
              scan1_output=m7.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & diet & strain) [blup values] (scan1blup): ",i)
              )
  plot(m7.snps[[i]]$lod, 
       m7.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m7.genes[[i]],
       main = paste0("SNP Association Mapping (sex & generation & diet & strain) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m8 (Sex + Diet + Strain)
```{r plot scan result for m8, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Diet & Strain (m8)")
load(paste0("output/qtl.mapping_sorted_all.grid/m8_sex.diet.strain_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m8.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m8_sex.diet.strain_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m8_sex.diet.strain_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m8.qtl.out[[i]]) # overall maximum LOD score
plot(m8.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & diet & strain) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m8_sex.diet.strain_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m8.gwas[[i]]$lod, 
     m8.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & diet & strain) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m8_sex.diet.strain_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m8_sex.diet.strain_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m8.qtl.out)){
  chr <- as.character(m8.peak[[i]][m8.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m8.coef[[i]], 
              map[chr], 
              scan1_output=m8.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & diet & strain) (scan1coef): ",i)
              )
  plot_coefCC(m8.blup[[i]], 
              map[chr], 
              scan1_output=m8.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & diet & strain) [blup values] (scan1blup): ",i)
              )
  plot(m8.snps[[i]]$lod, 
       m8.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m8.genes[[i]],
       main = paste0("SNP Association Mapping (sex & diet & strain) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m9 (Sex + Generation + Genotype)
```{r plot scan result for m9, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Generation & Genotype (m9)")
load(paste0("output/qtl.mapping_sorted_all.grid/m9_sex.generation.genotype_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m9.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m9_sex.generation.genotype_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m9_sex.generation.genotype_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m9.qtl.out[[i]]) # overall maximum LOD score
plot(m9.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & generation & genotype) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m9_sex.generation.genotype_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m9.gwas[[i]]$lod, 
     m9.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & generation & genotype) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m9_sex.generation.genotype_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m9_sex.generation.genotype_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m9.qtl.out)){
  chr <- as.character(m9.peak[[i]][m9.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m9.coef[[i]], 
              map[chr], 
              scan1_output=m9.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & genotype) (scan1coef): ",i)
              )
  plot_coefCC(m9.blup[[i]], 
              map[chr], 
              scan1_output=m9.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & genotype) [blup values] (scan1blup): ",i)
              )
  plot(m9.snps[[i]]$lod, 
       m9.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m9.genes[[i]],
       main = paste0("SNP Association Mapping (sex & generation & genotype) (scan1snps): ",i))
  
#}
#dev.off()
#}
```

### Plot Scan Results: m10 (Sex + Genotype)
```{r plot scan result for m10, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Genotype (m10)")
load(paste0("output/qtl.mapping_sorted_all.grid/m10_sex.genotype_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m10.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m10_sex.genotype_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m10_sex.genotype_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m10.qtl.out[[i]]) # overall maximum LOD score
plot(m10.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & genotype) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m10_sex.genotype_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m10.gwas[[i]]$lod, 
     m10.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & genotype) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m10_sex.genotype_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m10_sex.genotype_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m10.qtl.out)){
  chr <- as.character(m10.peak[[i]][m10.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m10.coef[[i]], 
              map[chr], 
              scan1_output=m10.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & genotype) (scan1coef): ",i)
              )
  plot_coefCC(m10.blup[[i]], 
              map[chr], 
              scan1_output=m10.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & genotype) [blup values] (scan1blup): ",i)
              )
  plot(m10.snps[[i]]$lod, 
       m10.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m10.genes[[i]],
       main = paste0("SNP Association Mapping (sex & genotype) (scan1snps): ",i))
  
#}
#dev.off()
#}
```

### Plot Scan Results: m11 (Sex + Generation + Diet + Genotype)
```{r plot scan result for m11, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Generation & Diet & Genotype (m11)")
load(paste0("output/qtl.mapping_sorted_all.grid/m11_sex.generation.diet.genotype_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m11.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m11_sex.generation.diet.genotype_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m11_sex.generation.diet.genotype_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m11.qtl.out[[i]]) # overall maximum LOD score
plot(m11.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & generation & diet & genotype) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m11_sex.generation.diet.genotype_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m11.gwas[[i]]$lod, 
     m11.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & generation & diet & genotype) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m11_sex.generation.diet.genotype_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m11_sex.generation.diet.genotype_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m11.qtl.out)){
  chr <- as.character(m11.peak[[i]][m11.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m11.coef[[i]], 
              map[chr], 
              scan1_output=m11.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & diet & genotype) (scan1coef): ",i)
              )
  plot_coefCC(m11.blup[[i]], 
              map[chr], 
              scan1_output=m11.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & diet & genotype) [blup values] (scan1blup): ",i)
              )
  plot(m11.snps[[i]]$lod, 
       m11.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m11.genes[[i]],
       main = paste0("SNP Association Mapping (sex & generation & diet & genotype) (scan1snps): ",i))
  
#}
#dev.off()
#}
```

### Plot Scan Results: m12 (Sex + Diet + Genotype)
```{r plot scan result for m12, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Diet & Genotype (m12)")
load(paste0("output/qtl.mapping_sorted_all.grid/m12_sex.diet.genotype_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m12.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m12_sex.diet.genotype_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m12_sex.diet.genotype_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m12.qtl.out[[i]]) # overall maximum LOD score
plot(m12.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & diet & genotype) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m12_sex.diet.genotype_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m12.gwas[[i]]$lod, 
     m12.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & diet & genotype) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m12_sex.diet.genotype_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m12_sex.diet.genotype_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m12.qtl.out)){
  chr <- as.character(m12.peak[[i]][m12.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m12.coef[[i]], 
              map[chr], 
              scan1_output=m12.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & diet & genotype) (scan1coef): ",i)
              )
  plot_coefCC(m12.blup[[i]], 
              map[chr], 
              scan1_output=m12.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & diet & genotype) [blup values] (scan1blup): ",i)
              )
  plot(m12.snps[[i]]$lod, 
       m12.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m12.genes[[i]],
       main = paste0("SNP Association Mapping (sex & diet & genotype) (scan1snps): ",i))
  
#}
#dev.off()
#}
```

### Plot Scan Results: m13 (Sex + Generation + Strain + Genotype)
```{r plot scan result for m13, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Generation & Strain & Genotype (m13)")
load(paste0("output/qtl.mapping_sorted_all.grid/m13_sex.generation.strain.genotype_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m13.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m13_sex.generation.strain.genotype_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m13_sex.generation.strain.genotype_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m13.qtl.out[[i]]) # overall maximum LOD score
plot(m13.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & generation & strain & genotype) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m13_sex.generation.strain.genotype_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m13.gwas[[i]]$lod, 
     m13.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & generation & strain & genotype) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m13_sex.generation.strain.genotype_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m13_sex.generation.strain.genotype_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m13.qtl.out)){
  chr <- as.character(m13.peak[[i]][m13.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m13.coef[[i]], 
              map[chr], 
              scan1_output=m13.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & strain & genotype) (scan1coef): ",i)
              )
  plot_coefCC(m13.blup[[i]], 
              map[chr], 
              scan1_output=m13.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & strain & genotype) [blup values] (scan1blup): ",i)
              )
  plot(m13.snps[[i]]$lod, 
       m13.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m13.genes[[i]],
       main = paste0("SNP Association Mapping (sex & generation & strain & genotype) (scan1snps): ",i))
  
#}
#dev.off()
#}
```

### Plot Scan Results: m14 (Sex + Strain + Genotype)
```{r plot scan result for m14, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Strain & Genotype (m14)")
load(paste0("output/qtl.mapping_sorted_all.grid/m14_sex.strain.genotype_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m14.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m14_sex.strain.genotype_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m14_sex.strain.genotype_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m14.qtl.out[[i]]) # overall maximum LOD score
plot(m14.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & strain & genotype) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m14_sex.strain.genotype_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m14.gwas[[i]]$lod, 
     m14.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & strain & genotype) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m14_sex.strain.genotype_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m14_sex.strain.genotype_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m14.qtl.out)){
  chr <- as.character(m14.peak[[i]][m14.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m14.coef[[i]], 
              map[chr], 
              scan1_output=m14.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & strain & genotype) (scan1coef): ",i)
              )
  plot_coefCC(m14.blup[[i]], 
              map[chr], 
              scan1_output=m14.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & strain & genotype) [blup values] (scan1blup): ",i)
              )
  plot(m14.snps[[i]]$lod, 
       m14.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m14.genes[[i]],
       main = paste0("SNP Association Mapping (sex & strain & genotype) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m15 (Sex + Generation + Diet + Strain + Genotype)
```{r plot scan result for m15, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Generation & Diet & Strain & Genotype (m15)")
load(paste0("output/qtl.mapping_sorted_all.grid/m15_sex.generation.diet.strain.genotype_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m15.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m15_sex.generation.diet.strain.genotype_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m15_sex.generation.diet.strain.genotype_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m15.qtl.out[[i]]) # overall maximum LOD score
plot(m15.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & generation & diet & strain & genotype) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m15_sex.generation.diet.strain.genotype_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m15.gwas[[i]]$lod, 
     m15.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & generation & diet & strain & genotype) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m15_sex.generation.diet.strain.genotype_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m15_sex.generation.diet.strain.genotype_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m15.qtl.out)){
  chr <- as.character(m15.peak[[i]][m15.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m15.coef[[i]], 
              map[chr], 
              scan1_output=m15.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & diet & strain & genotype) (scan1coef): ",i)
              )
  plot_coefCC(m15.blup[[i]], 
              map[chr], 
              scan1_output=m15.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & generation & diet & strain & genotype) [blup values] (scan1blup): ",i)
              )
  plot(m15.snps[[i]]$lod, 
       m15.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m15.genes[[i]],
       main = paste0("SNP Association Mapping (sex & genotype) (scan1snps): ",i))
  
#}
#dev.off()
#}

```

### Plot Scan Results: m16 (Sex + Diet + Strain + Genotype)
```{r plot scan result for m16, fig.height=10, fig.width=17, eval=TRUE}

print("Covariates: Sex & Diet & Strain & Genotype (m16)")
load(paste0("output/qtl.mapping_sorted_all.grid/m16_sex.diet.strain.genotype_",name.phenocsv,"_sorted_all.grid.RData"))

##############################################
## permutation results
##############################################
#The default is to return the 5% significance thresholds. Thresholds for other (or for multiple) significance levels can be obtained via the alpha argument.

#for(i in names(m16.qtl.out)){
#for(i in phenotypes){
operm <- get(load(paste0("output/permu_sorted_all.grid/m16_sex.diet.strain.genotype_nperm.100_",name.phenocsv,"_", i,"_sorted_all.grid.RData")))
cutoff = summary(operm, alpha = c(0.2, 0.1, 0.05, 0.01))
cutoff

##############################################
## Genome wide results
##############################################
#pdf(file = paste0("output/figures_sorted/m16_sex.diet.strain.genotype_",name.phenocsv,"_", i,"_genomescan_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
ymx <- maxlod(m16.qtl.out[[i]]) # overall maximum LOD score
plot(m16.qtl.out[[i]], 
  map, 
  lodcolumn=1,
  col="slateblue", 
  ylim=c(0, ymx*1.3),
  main=paste0("Genomescan (sex & diet & strain & genotype) (scan1): ", i))
  abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2) 
#dev.off()

#pdf(file = paste0("output/figures_sorted/m16_sex.diet.strain.genotype_",name.phenocsv,"_", i,"_gwas_sorted.pdf"), width = 14)
#par(mar=c(5.1, 4.1, 1.1, 1.1))
plot(m16.gwas[[i]]$lod, 
     m16.gwas[[i]]$snpinfo, 
     altcol="green4", 
     gap=0, 
     main = paste0("GWAS (sex & diet & strain & genotype) (scan1snps): ",i))
#     abline(h = cutoff, col = c("purple", "red", "blue", "green"), lwd = 2)
#dev.off()

##############################################
## QTL Results
##############################################

#load blup results
load(paste0("output/blup_sorted_all.grid/m16_sex.diet.strain.genotype_blup_",name.phenocsv,"_",i,"_sorted_all.grid.RData"))

#coeffects plot
#pdf(file = paste0("output/figures_sorted/m16_sex.diet.strain.genotype_",name.phenocsv,"_",i,"_coeff_geneplot_sorted.pdf"), width = 14)
#for(i in names(m16.qtl.out)){
  chr <- as.character(m16.peak[[i]][m16.peak[[i]]$lodcolumn == i,"chr"])
  par(mar=c(4.1, 4.1, 1.6, 1.6))
  plot_coefCC(m16.coef[[i]], 
              map[chr], 
              scan1_output=m16.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & diet & strain & genotype) (scan1coef): ",i)
              )
  plot_coefCC(m16.blup[[i]], 
              map[chr], 
              scan1_output=m16.qtl.out[[i]], 
              bgcolor="gray95", #legend="bottomleft", 
              main =paste0("Genomescan (sex & diet & strain & genotype) [blup values] (scan1blup): ",i)
              )
  plot(m16.snps[[i]]$lod, 
       m16.snps[[i]]$snpinfo, 
       drop_hilit=1.5, 
       genes=m16.genes[[i]],
       main = paste0("SNP Association Mapping (sex & diet & strain & genotype) (scan1snps): ",i))
  
#}
#dev.off()
#}

```